### PR TITLE
Add support for Angular v14

### LINKED ISSUE #12593: [REQ] [typescript-angular] Support Angular 14
Description
I would like to have the API clients generated using ngVersion>=14.0.0.

Suggest a fix/enhancement
Allow the ngVersion to be set to 14 and generate the code compatible with this Angular version.

### PR DESCRIPTION
**Pull request by marinus-suniram**
- Support Angular 14
- Support rxjs 7.5.5, tsickle 0.46.3, ngPackagr 14.0.2, zonejs 0.11.5
- Requires typescript >= 4.6.0 and < 4.8.0

Fixes  #12593 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.1.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

**Comment by wing328** [comment]
cc @TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @topce (2018/10) @akehir (2019/07) @petejohansonxo (2019/11) @amakhrov (2020/02) @davidgamero (2022/03) @mkusaka (2022/04)

**Review by macjohnny — COMMENTED** [review]
thanks for your contribution. can you please remove
- bin/configs/typescript-angular-v14-oneOf.yaml
- bin/configs/typescript-angular-v14-provided-in-any.yaml
- bin/configs/typescript-angular-v14-provided-in-root-with-npm.yaml
?

**Comment by tispBe** [comment]
Any update on this? We are currently blocked by the lack of Angular 14 support. Thanks a lot for your efforts!

**Comment by davide1995** [comment]
Any update guys? Seems like a step in the pipeline failed

**Comment by marinus-suniram** [comment]
> Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.
> 
> Let me know if you need help fixing it.
> 
> Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

Thanks I have update my git config.

File: bin/configs/typescript-angular-v14-provided-in-root.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+generatorName: typescript-angular
```
[the remark this answers is not in the document]
Comment (marinus-suniram) [reply bin/configs/typescript-angular-v14-provided-in-root.yaml:1]: Thx for the feedback. I'm happy to remove the test config, but is this not part of Angular 14 regression tests?

**Comment by Nesci28** [comment]
@marinus-suniram i think you must commit something the trigger the build with your new configs.

**Comment by wing328** [comment]
```
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   samples/server/petstore/aspnetcore-6.0-pocoModels/.openapi-generator/FILES

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	samples/client/petstore/typescript-angular-v14-oneOf/builds/default/.openapi-generator-ignore
	samples/client/petstore/typescript-angular-v14-oneOf/builds/default/.openapi-generator/
	samples/client/petstore/typescript-angular-v14-provided-in-any/builds/default/.openapi-generator-ignore
	samples/client/petstore/typescript-angular-v14-provided-in-any/builds/default/.openapi-generator/
	samples/client/petstore/typescript-angular-v14-provided-in-root/builds/default/.openapi-generator-ignore
	samples/client/petstore/typescript-angular-v14-provided-in-root/builds/default/.openapi-generator/
	samples/client/petstore/typescript-angular-v14-provided-in-root/builds/with-npm/.openapi-generator-ignore
	samples/client/petstore/typescript-angular-v14-provided-in-root/builds/with-npm/.openapi-generator/
```

can you please add these new files in the samples?

File: bin/configs/typescript-angular-v14-provided-in-root.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (macjohnny) [reply bin/configs/typescript-angular-v14-provided-in-root.yaml:1]: we can keep it and delete it for older angular versions if you want. this way, we have less code to review when something changes. and these type of variations are not version-specific.

to enable the build/test to be run as part of the CI, you need to add it here:

https://github.com/OpenAPITools/openapi-generator/blob/186ad25cee158f6ac9b7d22448cbbdd1db065178/pom.xml#L1212

[the remark this answers is not in the document]
Comment (marinus-suniram) [reply bin/configs/typescript-angular-v14-provided-in-root.yaml:1]: Ok, I have removed 

bin/configs/typescript-angular-v14-oneOf.yaml
bin/configs/typescript-angular-v14-provided-in-any.yaml
bin/configs/typescript-angular-v14-provided-in-root-with-npm.yaml

File: samples/client/petstore/typescript-angular-v14-provided-in-root/pom.xml [hunk 2]
```diff
@@ -0,0 +1,140 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>org.openapitools</groupId>
+    <artifactId>typescript-angular-v13-provided-in-root-tests</artifactId>
+    <packaging>pom</packaging>
+    <version>1.0-SNAPSHOT</version>
+
+    <name>Typescript-Angular v13 Petstore Client</name>
```
Comment (macjohnny) [inline samples/client/petstore/typescript-angular-v14-provided-in-root/pom.xml:9]:
```suggestion
    <name>Typescript-Angular v14 Petstore Client</name>
```

File: samples/client/petstore/typescript-angular-v14-provided-in-root/pom.xml [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-clean-plugin</artifactId>
+                <version>3.1.0</version>
+
+                <executions>
+                    <execution>
+                        <id>clean-typescript-angular-v13-test-outputs</id>
```
Comment (macjohnny) [inline samples/client/petstore/typescript-angular-v14-provided-in-root/pom.xml:62]:
```suggestion
                        <id>clean-typescript-angular-v14-test-outputs</id>
```

File: samples/client/petstore/typescript-angular-v14-provided-in-root/pom.xml [hunk 4]
```diff
@@ -0,0 +1,140 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>org.openapitools</groupId>
+    <artifactId>typescript-angular-v13-provided-in-root-tests</artifactId>
```
Comment (macjohnny) [inline samples/client/petstore/typescript-angular-v14-provided-in-root/pom.xml:5]:
```suggestion
    <artifactId>typescript-angular-v14-provided-in-root-tests</artifactId>
```

**Review by macjohnny — APPROVED** [review]
LGTM

**Comment by macjohnny** [comment]
@marinus-suniram can you please merge the most recent master, re-generate the samples, and add the new tests to
https://github.com/OpenAPITools/openapi-generator/blob/186ad25cee158f6ac9b7d22448cbbdd1db065178/pom.xml#L1212

?

**Comment by macjohnny** [comment]
unfortunately, there are still outdated samples, can you please check again?

**Comment by KL2808** [comment]
# Heading 1
## Heading 2
blablabla

Vorname | Nachname
---|---
Kevin | Lieder
Lukas | Walter
Kevin | Lieder
Lukas | Walter
Kevin | Lieder
Lukas | Walter
Kevin | Lieder
Lukas | Walter

**Comment by Nesci28** [comment]
@marinus-suniram can we do something to help ?

**Comment by davide1995** [comment]
Thanks for the merge. When will the new version be released?

**Comment by wing328** [comment]
This week or next as we've a few changes to be include in the upcoming release.

**Comment by davide1995** [comment]
Is it not possible in the meantime to use a snapshot version with gradle?

**Pull request merged**